src/extremeweatherbench/metrics.py

```diff
@@ -135,31 +146,42 @@ def compute(self, forecast: xr.DataArray, observation: xr.DataArray):
                 )
                 # Ensure that the forecast has a full day of data for each day
                 # after centering on the max of min timestamp
-                filtered_forecast = utils.truncate_incomplete_days(filtered_forecast)
-                lead_time = filtered_forecast.where(
-                    filtered_forecast.time == max_min_timestamp, drop=True
-                ).lead_time
-                filtered_forecast_max_min = filtered_forecast.where(
-                    filtered_forecast
-                    == filtered_forecast.groupby("time.dayofyear").min().max(),
-                    drop=True,
-                )
-                # TODO: add temporal displacement error, which is
-                # filtered_forecast_max_min.time.values[0] - max_min_timestamp
-                if max_min_timestamp in filtered_forecast.time.values:
-                    max_min_mae_dataarray = xr.DataArray(
-                        data=abs(filtered_forecast_max_min - max_min_value),
-                        dims=["lead_time"],
-                        coords={"lead_time": lead_time.values},
-                        attrs={
-                            "description": (
-                                "Mean absolute error of forecasted highest minimum temperature values,"
-                                "where lead_time is the time from initialization until the highest minimum"
-                                "observed temperature."
-                            )
-                        },
+                if filtered_forecast.time.shape[0] == 0:
+                    logger.info(
+                        "Init time %s insufficient data for max of min temp",
+                        pd.to_datetime(init_time.values),
                     )
-                    max_min_mae_values.append(max_min_mae_dataarray)
+                else:
+                    filtered_forecast = utils.truncate_incomplete_days(
+                        filtered_forecast
+                    )
+                    lead_time = filtered_forecast.where(
+                        filtered_forecast.time == max_min_timestamp, drop=True
+                    ).lead_time
+                    filtered_forecast_max_min = filtered_forecast.where(
+                        filtered_forecast
+                        == filtered_forecast.groupby("time.dayofyear")
+                        .min()
+                        .max(),  # TODO check if "time" "dayofyear" as it is in point obs works here
+                        drop=True,
+                    )
+
+                    # TODO: add temporal displacement error, which is
+                    # filtered_forecast_max_min.time.values[0] - max_min_timestamp
+                    if max_min_timestamp in filtered_forecast.time.values:
```

Comment:
By my count, we're three-levels deep with this `if` statement. What is going on here, and why do we need such deeply nested code? This is usually a sign that we should invert the logic in our if/else's and employ short-circuiting, or re-factor some of this logic wholesale to utility functions. Code this deeply nested is very hard to follow because you have to keep multiple, prior conditionals and their context in mind, which is quite difficult in practice.